test/e2e/utils_test.go

```diff
@@ -32,25 +32,125 @@ import (
+						},
+						{
+							Type:               string(clusterv1beta1.AgentHealthy),
+							LastTransitionTime: metav1.Now(),
+							ObservedGeneration: 0,
+							Status:             metav1.ConditionTrue,
+							Reason:             "HealthyCluster",
+							Message:            "set to be healthy",
+						},
+					},
+					LastReceivedHeartbeat: metav1.Now(),
+				},
 			},
-			Spec: clusterv1beta1.MemberClusterSpec{
-				Identity: rbacv1.Subject{
-					Name:      hubClusterSAName,
-					Kind:      "ServiceAccount",
-					Namespace: fleetSystemNS,
+		}
+
+		return hubClient.Status().Update(ctx, imcObj)
+	}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to mark member cluster as healthy")
+}
+
+// markMemberClusterAsUnhealthy marks the specified member cluster as unhealthy (last
+// received heartbeat expired).
+func markMemberClusterAsUnhealthy(name string) {
+	Eventually(func() error {
+		mcObj := &clusterv1beta1.MemberCluster{}
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: name}, mcObj); err != nil {
+			return err
+		}
+
+		mcObj.Status = clusterv1beta1.MemberClusterStatus{
+			AgentStatus: []clusterv1beta1.AgentStatus{
+				{
+					Type: clusterv1beta1.MemberAgent,
+					Conditions: []metav1.Condition{
+						{
+							Type:               string(clusterv1beta1.AgentJoined),
+							LastTransitionTime: metav1.NewTime(time.Now().Add(-time.Hour * 24)),
+							ObservedGeneration: 0,
+							Status:             metav1.ConditionTrue,
+							Reason:             "JoinedCluster",
+							Message:            "set to be joined",
+						},
+						{
+							Type:               string(clusterv1beta1.AgentHealthy),
+							LastTransitionTime: metav1.NewTime(time.Now().Add(-time.Hour * 24)),
+							ObservedGeneration: 0,
+							Status:             metav1.ConditionTrue,
+							Reason:             "HealthyCluster",
+							Message:            "set to be healthy",
+						},
+					},
+					LastReceivedHeartbeat: metav1.NewTime(time.Now().Add(-time.Hour * 24)),
 				},
 			},
 		}
-		Expect(hubClient.Create(ctx, mcObj)).To(Succeed(), "Failed to create member cluster object")
+		return hubClient.Status().Update(ctx, mcObj)
+	}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to mark member cluster as unhealthy")
+}
+
+// markMemberClusterAsLeft marks the specified member cluster as left.
+func markMemberClusterAsLeft(name string) {
+	mcObj := &clusterv1beta1.MemberCluster{}
+	Eventually(func() error {
+		// Add a custom deletion blocker finalizer to the member cluster.
+		if err := hubClient.Get(ctx, types.NamespacedName{Name: name}, mcObj); err != nil {
+			return err
+		}
+
+		mcObj.Finalizers = append(mcObj.Finalizers, customDeletionBlockerFinalizer)
+		return hubClient.Update(ctx, mcObj)
+	}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to mark member cluster as left")
+
+	Expect(client.IgnoreNotFound(hubClient.Delete(ctx, mcObj))).To(Succeed(), "Failed to delete member cluster")
```

Comment:
if we add the finalizer, it will never be 404?

Reply:
Done.